geonode_mapstore_client/client/js/actions/gnresource.js

```diff
@@ -114,3 +115,18 @@ export function setResourcePermissions(permissions) {
         permissions
     };
 }
+
+/**
+* Set resource permissions
+* @memberof actions.gnresource
+* @param {object} permissions permissions info
+* @param {bool} permissions.canEdit can edit permission
+* @param {bool} permissions.canView can view permission
+*/
+
+export function setUserResourcePermissions(permissions) {
```

Comment:
we are storing the permission variable as `selectedLayerPermissions` so it shoud be better to call this action with 
`setSelectedLayerPermissions` with action `SET_SELECTED_LAYER_PERMISSIONS`. Please replace this also in all other files.